src/components/Delegates/DelegateCardList/DelegateCardList.tsx

```diff
@@ -45,34 +43,24 @@ export default function DelegateCardList({
     setMeta(initialDelegates.meta);
   }, [initialDelegates]);
 
-  const handleClick = (
-    e: React.MouseEvent<HTMLDivElement, MouseEvent>,
-    href: string
-  ) => {
-    e.preventDefault();
-    router.push(href);
-  };
-
-  const loadMore = async (page: any) => {
+  const loadMore = async () => {
     if (!fetching.current && meta.hasNextPage) {
       fetching.current = true;
-      const data = await fetchDelegates(page);
-      const existingIds = new Set(
-        pages.flatMap((page) => page.delegates.map((d) => d.address))
-      );
+      const data = await fetchDelegates(meta.currentPage + 1);
+
+      // TODO: This is a temporary fix to avoid duplicates when filtering by "Weighted random"
```

Comment:
or actually, seems like the issue is here: https://github.com/voteagora/agora-next/blob/andrei/agora-1601-misc-improvements-cleanup/src/app/delegates/page.jsx#L63
we do pass the `seed` down to this component but we don't actually set it in the `initialDelegates` load

Reply:
@stepandel, this might be a misunderstanding on my part, but I've been seeing dups in all paginated responses. In other words, each page contained at least a few items previously returned when the 'weighted random' filter is applied.